thrifty-java-codegen/src/main/kotlin/com/microsoft.thrifty.gen/ThriftyCodeGenerator.kt

```diff
@@ -622,7 +628,18 @@ class ThriftyCodeGenerator {
 
             // Read
             read.beginControlFlow("case \$L:", field.id)
-            GenerateReaderVisitor(typeResolver, read, fieldName, field.type.trueType).generate()
+            val useDefaultEnumValue = tt.isEnum && acceptUnknownEnumValues && !field.required
+            object : GenerateReaderVisitor(typeResolver, read, fieldName, field.type.trueType, useDefaultEnumValue) {
```

Comment:
What do you think about pushing this logic down into `GenerateReaderVisitor.useReadValue` instead of doing an anonymous subclass here?

Reply:
I was trying out a few other things which required access to the `field` directly. But eventually ended up removing that access completely. This can absolutely be moved now